### PR TITLE
89 save checking

### PR DESCRIPTION
**Pull request by SamiFawcett**
Prompts the user to save their file if the changes they made haven't been saved at the time of closing the application window.

### CODE REVIEW

File: src/main/java/com/ezasm/gui/WindowCloseListener.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    public WindowCloseListener() {
+        super();
+    }
+
+    @Override
+    public void windowClosing(WindowEvent e) {
+        if (!Window.getInstance().getEditor().getFileSaved()) {
+            boolean resp = promptYesNoDialog("Exiting...",
+                    "Your changes have not been saved.\nWould you like to save them?");
```
Comment (WyattRoss) [inline src/main/java/com/ezasm/gui/WindowCloseListener.java:19]: This isn't fully platform independent. Not all platforms use `\n` for newline. To get a system dependent newline character use `System.getProperties("line.separator")`

**Review by WyattRoss — CHANGES_REQUESTED** [review]
Other than the code level changes described in other comments, there are a few behavioral changes I'd like:
- If you open the program you're presented with an empty file. I'd like to be able to close the program without being prompted to save an empty file
- Add a cancel button to the save dialog, pressing it should neither save the file nor close the program, but return the user to the editor
- This one is a bit harder to implement, and I understand if you don't, but making it such that modifying the file, then undoing those modifications (either manually, or with \<C-z\>) shouldn't prompt a save dialog

There is also one bug:
When you make a change to the file, then save that file, pressing ctrl+z to undo the last change will erase the contents of the file.

**Comment by SamiFawcett** [comment]
I will setup issues for the problems I couldn't solve for this pr.

**Review by WyattRoss — APPROVED** [review]
Send it

**Pull request merged**